theano/compile/mode.py

```diff
@@ -111,6 +111,8 @@ class AddDestroyHandler(gof.Optimizer):
 
     """
     def apply(self, fgraph):
+        if theano.config.cycle_detection == 'fast':
```

Comment:
Why this change? I didn't expected this PR to change the tests that fail with cycle_detection=False. I would not be surprised that this change isn't needed and removing it "fix" the failing tests.